core/consensus/component.go

```diff
@@ -42,8 +42,8 @@ const (
 	protocolID = "/charon/consensus/qbft/1.0.0"
 )
 
-// NewComponent returns a new consensus QBFT component.
-func NewComponent(tcpNode host.Host, peers []p2p.Peer, p2pKey *ecdsa.PrivateKey) (*Component, error) {
```

Comment:
NewComponent looks more relevant 🙃

Reply:
😉

Reply:
if we name components "component", everything would be named component, that is why we should avoid those types of generics in naming